src/renderer/components/node/NodeOutputs.tsx

```diff
@@ -12,6 +13,14 @@ interface NodeOutputsProps {
 
 export const NodeOutputs = memo(({ outputs, id, schemaId }: NodeOutputsProps) => {
     const { functionDefinitions } = useContext(GlobalContext);
+    const useOutputDataContext = useContextSelector(ExecutionContext, (c) => c.useOutputData);
+
+    const useOutputData = useCallback(
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        (outputId: OutputId) => useOutputDataContext(id, outputId),
+        [useOutputDataContext, id]
+    );
```

Comment:
I added the `useInputData` "hook" for inputs because we have multiple input components. It was a means to avoid code duplication.

However, I don't think that this is worth it. It just adds more code. Do we expect to add more output components in the near future? If no, then this code should live in `GenericOutput`.

Reply:
Yes I plan on adding more output components

Reply:
Alright, then let's keep it. (Could you resolve this thread? I don't have the permissions to do so.)